packages/cli-kit/src/private/node/ui/components/ConcurrentOutput.test.tsx

```diff
@@ -0,0 +1,60 @@
+import ConcurrentOutput from './ConcurrentOutput.js'
+import {Signal} from '../../../../abort.js'
+import {unstyled} from '../../../../output.js'
+import React from 'react'
+import {describe, expect, test} from 'vitest'
+import {AbortController} from 'abort-controller'
+import {render} from 'ink-testing-library'
+import {Writable} from 'node:stream'
+
+describe('ConcurrentOutput', () => {
+  test('renders a stream of concurrent outputs from sub-processes', async () => {
+    // Given
+    let backendPromiseResolve: () => void
+
+    const backendPromise = new Promise<void>(function (resolve, _reject) {
+      backendPromiseResolve = resolve
+    })
+
+    const backendProcess = {
+      prefix: 'backend',
+      action: async (stdout: Writable, _stderr: Writable, _signal: Signal) => {
+        stdout.write('first backend message')
+        stdout.write('second backend message')
+        stdout.write('third backend message')
+
+        backendPromiseResolve()
+      },
+    }
+
+    const frontendProcess = {
+      prefix: 'frontend',
+      action: async (stdout: Writable, _stderr: Writable, _signal: Signal) => {
+        await backendPromise
+
+        stdout.write('first frontend message')
+        stdout.write('second frontend message')
+        stdout.write('third frontend message')
+      },
+    }
+    // When
+
+    const {lastFrame} = render(
+      <ConcurrentOutput processes={[backendProcess, frontendProcess]} abortController={new AbortController()} />,
+    )
+
+    // wait for all output to be rendered
+    await new Promise((resolve) => setTimeout(resolve, 100))
```

Comment:
This one can become a source of flakiness fairly quickly. Doesn't Ink provide a callback to know when the render has happened?